target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8064-unifi-ap-hd.dts

```diff
@@ -0,0 +1,342 @@
+
+			partitions {
+				compatible = "fixed-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partition@0 {
+					label = "SBL1";
+					reg = <0x0 0x20000>;
+					read-only;
+				};
+
+				partition@20000 {
+					label = "MIBIB";
+					reg = <0x20000 0x10000>;
+					read-only;
+				};
+
+				partition@30000 {
+					label = "SBL2";
+					reg = <0x30000 0x20000>;
+					read-only;
+				};
+
+				partition@50000 {
+					label = "SBL3";
+					reg = <0x50000 0x30000>;
+					read-only;
+				};
+
+				partition@80000 {
+					label = "DDRCONFIG";
+					reg = <0x80000 0x10000>;
+					read-only;
+				};
+
+				partition@90000 {
+					label = "SSD";
+					reg = <0x90000 0x100000>;
+					read-only;
+				};
+
+				partition@a0000 {
+					label = "TZ";
+					reg = <0xa0000 0x30000>;
+					read-only;
+				};
+
+				partition@d0000 {
+					label = "RPM";
+					reg = <0xd0000 0x20000>;
+					read-only;
+				};
+
+				partition@f0000 {
+					label = "APPSBL";
+					reg = <0xf0000 0xc0000>;
+					read-only;
+				};
+
+				partition@1b0000 {
+					label = "APPSBLENV";
+					reg = <0x1b0000 0x10000>;
+					read-only;
+				};
+
+				eeprom: partition@1c0000 {
+					label = "EEPROM";
+					reg = <0x1c0000 0x10000>;
+					read-only;
+				};
+
+				partition@1d0000 {
+					label = "bootselect";
+					reg = <0x1d0000 0x10000>;
+				};
+
+				partition@1e0000 {
+					compatible = "denx,uimage";
+					label = "firmware";
+					reg = <0x1e0000 0xe70000>;
+				};
+
+				partition@1050000 {
+					label = "kernel1";
+					reg = <0x1050000 0xe70000>;
+					read-only;
+				};
+
+				partition@1ec0000 {
+					label = "debug";
+					reg = <0x1ec0000 0x100000>;
+					read-only;
+				};
+
+				partition@1fc0000 {
+					label = "cfg";
+					reg = <0x1fc0000 0x40000>;
+					read-only;
+				};
+			};
+		};
+	};
+};
+
+&adm_dma {
+	status = "okay";
+};
+
+&nand_controller {
+	status = "okay";
+
+	pinctrl-0 = <&nand_pins>;
+	pinctrl-names = "default";
+	nand-ecc-strength = <4>;
+	nand-bus-width = <8>;
+};
+
+&mdio0 {
+	status = "okay";
+
+	pinctrl-0 = <&mdio0_pins>;
+	pinctrl-names = "default";
+
+	phy4: ethernet-phy@4 {
+		device_type = "ethernet-phy";
+		reg = <4>;
+	};
+
+	phy5: ethernet-phy@5 {
+		device_type = "ethernet-phy";
+		reg = <5>;
+	};
+};
+
+&gmac2 {
+	status = "okay";
+
+	phy-mode = "sgmii";
+	qcom,id = <2>;
+	qcom,emulation = <0>;
+	mdiobus = <&mdio0>;
+
+	mtd-mac-address = <&eeprom 0x0>;
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&gmac1 {
+	status = "okay";
+
+	phy-mode = "sgmii";
+	qcom,id = <1>;
+	qcom,pcs-chanid = <1>;
+	qcom,emulation = <0>;
+	mdiobus = <&mdio0>;
+
+	mtd-mac-address = <&eeprom 0x6>;
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&pcie0 {
+	status = "okay";
+
+	bridge@0,0 {
+		reg = <0x00000000 0 0 0 0>;
+		#address-cells = <3>;
+		#size-cells = <2>;
+		ranges;
+
+		wifi@1,0 {
+			compatible = "pci168c,0046";
+			reg = <0x00010000 0 0 0 0>;
+
+			mtd-mac-address = <&eeprom 0x6>;
```

Comment:
I've copied these nodes over from r7800 dts (same chipsets) and changed the mac location, unfortunately without any effect. Also tried with `compatible = "qcom,ath10k"`, also not working...

Reply:
`compatible = "qcom,ath10k";` is the correct one but its mostly a formality

I don't know what the "bridge" node is for, maybe try without that...at this point a full kernel log would help (link to a gist to save space

Reply:
also you can poke around the /sys pseudo-filesystem
to see whats there and compare devices IDs
and whether or not there is a root bus and what address it is...

`/sys/class/pci_bus`
`/sys/bus/pci/devices`
pci* in `/sys/devices`

Reply:
Apparently `mtd-mac-address` DTS property does not work with ath10k, so I'm focusing on the `10_fix_wifi_mac` approach for now...

Reply:
https://github.com/openwrt/openwrt/pull/3403

Reply:
Doesn't wort even with the patch from your linked PR. I get the impression setting the WLAN MAC for ath10k can only work when manually reading/supplying caldata...

Reply:
Is it possible you have the wrong `reg` value? Should it not be like this?

(I am not certain on where the reg values come from but I think kernel log provides...)

```
[    0.533154] pci 0000:01:00.0: [168c:0046] type 00 class 0x028000
[    0.773035] pci 0001:01:00.0: [168c:0046] type 00 class 0x028000

```

```
&pcie0 {
	status = "okay";

	bridge@0,0,0 {
		reg = <0x00000000 0 0 0 0>;
		#address-cells = <3>;
		#size-cells = <2>;
		ranges;

		wifi@0,1,0 {
			compatible = "qcom,ath10k";
			reg = <0x00000000 1 0 0 0>;

			mtd-mac-address = <&eeprom 0x6>;
			mtd-mac-address-increment = <1>;
		};
	};
};

&pcie1 {
	status = "okay";
	max-link-speed = <1>;

	bridge@1,0,0 {
		reg = <0x00000001 0 0 0 0>;
		#address-cells = <3>;
		#size-cells = <2>;
		ranges;

		wifi@1,1,0 {
			compatible = "qcom,ath10k";
			reg = <0x00000001 1 0 0 0>;

			mtd-mac-address = <&eeprom 0x6>;
			mtd-mac-address-increment = <2>;
		};
	};
};
```